### PR TITLE
Other miscellaneous fixes

### PR DESCRIPTION
**Pull request by debora-pe**
These are 2 miscellaneous fixes/changes:

1. Add an error message in the case when the .pypeit file does not include any science/standard frames and `run_pypeit` is run without the `-c` flag. The current behavior is to just exit the reduction after a few seconds saying "Data reduction complete", but not giving any other info.

2. A diagnostic table about the wavelength calibration is added to the the spec2d datamodel. This is useful in the context of the ADAP, but it is also vey helpful in general for PypeIt users.

### CODE REVIEW

**Comment by debora-pe** [comment]
Run the tests and all look good:
```

Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  191 passed, 1274 warnings in 159.66s (0:02:39) ---
--- PYTEST UNIT TESTS PASSED  112 passed, 1717 warnings in 829.83s (0:13:49) ---
--- PYTEST VET TESTS PASSED  24 passed, 1571 warnings in 840.84s (0:14:00) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 152/152 TESTS  ---
Testing Started at 2022-06-23T16:51:04.112140
Testing Completed at 2022-06-23T23:41:48.346924
Total Time: 6:50:44.234784
```

File: pypeit/spec2dobj.py
Comment (profxj) [inline pypeit/spec2dobj.py:81]: Please update the version number

File: pypeit/wavecalib.py
Comment (profxj) [inline pypeit/wavecalib.py:309]: How about adding a couple more diagnostics?

- The minimum and maximum pixel of lines ID'd and fitted (gives a sense of how much extrapolation)

**Review by profxj — APPROVED** [review]
Am approving with a few minor suggestions/requests.

File: pypeit/wavecalib.py
Comment (profxj) [reply pypeit/wavecalib.py:309]: And I took the initiative to add `fwhm` which we've been thinking about including for some time.

Comment (profxj) [reply pypeit/wavecalib.py:309]: Actually, I am making that a side PR into this branch

File: pypeit/spec2dobj.py
Comment (debora-pe) [reply pypeit/spec2dobj.py:81]: done

File: pypeit/wavecalib.py
Comment (debora-pe) [reply pypeit/wavecalib.py:309]: added min max wavelength of lines ID'd and fitted. Also added the wavelength coverage (%) of those line.

File: pypeit/core/wavecal/autoid.py
Comment (profxj) [inline pypeit/core/wavecal/autoid.py]: Got myself confused on binning when I was thinking about this before.

Is the FWHM reported in the binning of the arc, native pixels, or the binning of the science?   Let's specify in the docs (and in `WaveFit`).  Thx!

File: pypeit/core/wavecal/wv_fitting.py
Comment (profxj) [inline pypeit/core/wavecal/wv_fitting.py]: As above, we should specify here whether it is native pixels and whether it is the Arc's pixels or the science  (as their binning can be different).

File: pypeit/core/wavecal/autoid.py
Comment (profxj) [inline pypeit/core/wavecal/autoid.py]: Thanks for pulling this out.  Next question -- can we call it from the other methods (at least `reidentify`) so that we have `measured_fwhm` for those spectrographs (mainly the echellettes)?

File: CHANGES.rst
Comment (kbwestfall) [inline CHANGES.rst:11]: Please merge in `develop` and `release`.

File: doc/master_wvcalib.rst
Comment (kbwestfall) [inline doc/master_wvcalib.rst]: We might want to have this generated in the docs automatically in case of format changes.  See `doc/scripts/make_example_files.py`.  Doesn't need to be for this PR, and I'm happy to help.

File: pypeit/core/wavecal/autoid.py
Comment (kbwestfall) [inline pypeit/core/wavecal/autoid.py]: Following on @profxj's comments, I'm unsure about this conversion, and we should be clear on this in the docs.  If the input spectrum is binned and `binspectral` is the binning of that input spectrum, is this converting the FWHM from the number of binned pixels to the number of unbinned pixels?  If so, shouldn't we be multiplying instead of dividing?

File: pypeit/scripts/chk_wavecalib.py
Comment (kbwestfall) [inline pypeit/scripts/chk_wavecalib.py]: Maybe be more explicit in the error message?  Something like: "Unrecognized file type.  Must be a MasterWaveCalib or spec2d file."

File: pypeit/scripts/parse_slits.py
Comment (kbwestfall) [inline pypeit/scripts/parse_slits.py:57]: Thanks for addressing the TODO!

**Review by kbwestfall — CHANGES_REQUESTED** [review]
Thanks, @deborape !  I'm not approving yet because I'd like to make sure the docs are clear on the binning and, specifically, whether the returned FWHM is in binned pixels or unbinned pixels.

File: pypeit/core/wavecal/autoid.py
Comment (debora-pe) [reply pypeit/core/wavecal/autoid.py]: @kbwestfall @profxj  thank you for bringing up this confusion about the `fwhm`. Now I am confused too.  I think got confused by the fact that in the doc it says that `['calibrations']['wavelengths']['fwhm']` should be in unbinned pixels, but looking at the code (at least `full_template`) it is not true. In `full_template` the input spectrum is used as it is (binned) and the fwhm is used as it is (which should apparently be binned). I'll check the other methods to be sure, and if I'm correct I'll fix this and the docs.

File: pypeit/scripts/chk_wavecalib.py
Comment (debora-pe) [reply pypeit/scripts/chk_wavecalib.py]: Done. Thanks!

File: pypeit/core/wavecal/autoid.py
Comment (debora-pe) [reply pypeit/core/wavecal/autoid.py]: I have checked and it seems fwhm is always used in binned pixels of the input arc frames. I corrected `measured_fwhm` and the docs.

File: CHANGES.rst
Comment (debora-pe) [reply CHANGES.rst:11]: I merged `develop`. I'll merge `release` after you guys take a look at these changes.

File: pypeit/core/wavecal/autoid.py
Comment (debora-pe) [reply pypeit/core/wavecal/autoid.py]: I have added `measured_fwhm` for the `reidentify` method.  The use of `measured_fwhm` is contingent to having the parameter `fwhm_fromlines` set to `True`, which is currently the default only for DEIMOS and MOSFIRE.

File: pypeit/core/wavecal/wv_fitting.py
Comment (debora-pe) [reply pypeit/core/wavecal/wv_fitting.py]: Done

File: pypeit/core/wavecal/autoid.py
Comment (debora-pe) [reply pypeit/core/wavecal/autoid.py]: Done

File: doc/master_wvcalib.rst
Comment (debora-pe) [reply doc/master_wvcalib.rst]: Thanks for the suggestion and the offer to help. I'll prefer to postpone this for a later PR.

File: pypeit/core/wavecal/autoid.py
Comment (kbwestfall) [reply pypeit/core/wavecal/autoid.py]: A couple lingering questions:

- This function no longer requires `binspectral` as an argument, right?
- I just want to confirm that you've checked a case when the binning of the arcs and science frames are different.  Is everything propagating as it used to?

Comment (debora-pe) [reply pypeit/core/wavecal/autoid.py]: @kbwestfall, for your first point, your right! I forgot that. Thanks. I'll fix it.

For the second point, the part of the code that I'm touching is only related to the FWHM that is used to identify arc lines. In this case, I'm pretty sure the arc spectrum is used as it is (i.e., with whatever binning it has), therefore the FWHM should also be expressed in binned pixels of the arc. After this step, I'm not really sure how the code handles cases when the binning of the arcs and science frames are different, but it should not affected by the changes I made here.

**Review by kbwestfall — APPROVED** [review]
Thanks for addressing all my comments!  If you think it's worth rerunning the tests before merging, please do so.  Regardless, I'm approving now.

**Comment by debora-pe** [comment]
All the tests pass.

```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  197 passed, 1277 warnings in 161.06s (0:02:41) ---
--- PYTEST UNIT TESTS PASSED  115 passed, 1862 warnings in 835.64s (0:13:55) ---
--- PYTEST VET TESTS PASSED  24 passed, 1571 warnings in 831.64s (0:13:51) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 152/152 TESTS  ---
Testing Started at 2022-07-08T17:16:20.414614
Testing Completed at 2022-07-09T02:05:01.399315
Total Time: 8:48:40.984701
```

**Comment by debora-pe** [comment]
merging

**Pull request merged**
